File: src/main/java/org/trp/shincolle/init/ModLootModifiers.java
package org.trp.shincolle.init;

import com.mojang.serialization.MapCodec;
import net.neoforged.neoforge.common.loot.IGlobalLootModifier;
import net.neoforged.neoforge.registries.DeferredRegister;
import org.trp.shincolle.Shincolle;
import org.trp.shincolle.loot.LegacyChestLootModifier;

public class ModLootModifiers {
    public static final DeferredRegister<MapCodec<? extends IGlobalLootModifier>> LOOT_MODIFIERS =
            DeferredRegister.create(net.neoforged.neoforge.registries.NeoForgeRegistries.Keys.GLOBAL_LOOT_MODIFIER_SERIALIZERS, Shincolle.MODID);

    public static final net.neoforged.neoforge.registries.DeferredHolder<MapCodec<? extends IGlobalLootModifier>, MapCodec<LegacyChestLootModifier>> LEGACY_CHEST_LOOT =
            LOOT_MODIFIERS.register("legacy_chest_loot", () -> LegacyChestLootModifier.CODEC);

    private ModLootModifiers() {}
}

File: src/main/java/org/trp/shincolle/init/ModLootModifiers.kt
package org.trp.shincolle.init

import com.mojang.serialization.MapCodec
import net.neoforged.neoforge.common.loot.IGlobalLootModifier
import net.neoforged.neoforge.registries.DeferredHolder
import net.neoforged.neoforge.registries.DeferredRegister
import net.neoforged.neoforge.registries.NeoForgeRegistries
import org.trp.shincolle.Shincolle
import org.trp.shincolle.loot.LegacyChestLootModifier
import java.util.function.Supplier

object ModLootModifiers {
    val LOOT_MODIFIERS: DeferredRegister<MapCodec<out IGlobalLootModifier?>?> =
        DeferredRegister.create<MapCodec<out IGlobalLootModifier?>?>(
            NeoForgeRegistries.Keys.GLOBAL_LOOT_MODIFIER_SERIALIZERS, Shincolle.MODID
        )

    val LEGACY_CHEST_LOOT: DeferredHolder<MapCodec<out IGlobalLootModifier?>?, MapCodec<LegacyChestLootModifier?>?> =
        LOOT_MODIFIERS.register<MapCodec<LegacyChestLootModifier?>?>(
            "legacy_chest_loot",
            Supplier { LegacyChestLootModifier.CODEC })
}
